swift/extractor/SwiftDispatcher.h

```diff
@@ -100,12 +109,12 @@ class SwiftDispatcher {
   // it actually gets emitted to handle recursive cases such as recursive calls, or recursive type
   // declarations
   template <typename E>
-  TrapLabel<ToTag<E>> assignNewLabel(E* e) {
-    assert(waitingForNewLabel == getCanonicalPointer(e) && "assignNewLabel called on wrong entity");
-    auto label = getLabel<ToTag<E>>();
+  TrapLabelOf<E> assignNewLabel(E* e) {
+    assert(waitingForNewLabel == Store::Handle{e} && "assignNewLabel called on wrong entity");
+    auto label = getLabel<TrapTagOf<E>>();
     trap.assignStar(label);
     store.insert(e, label);
-    waitingForNewLabel = nullptr;
+    waitingForNewLabel = {};
```

Comment:
Optional: As above, I'd prefer this to spell out `monostate` rather than relying on it being the first member.

Reply:
on the other hand, this is the documented main use case of `std::monospace` https://en.cppreference.com/w/cpp/utility/variant/monostate but I'm ok with being more explicit.